app/services/get_products_service.rb

```diff
@@ -39,6 +39,31 @@ def build_query
              tags.name AS tag_name, tags.color AS tag_color')
   end
 
+  def fetch_products
+    products = build_query
+    products.group_by { |product| product[:category_name] }.map do |category_name, elements|
+      {
+        name: category_name,
+        products: elements.map do |element|
```

Comment:
check the difference between map and each, and I don't think it is necessary to create map_product method here